### PR TITLE
Run processes as unprivileged user and simplify logging

### PR DESCRIPTION
**Pull request by jirutka**
Note: clown-client still run as root, because it starts bc-bridge that needs access to /dev/rawhidX which has perms 700. :(

### CODE REVIEW

File: etc/supervisord.conf [hunk 1]
```diff
@@ -1,85 +1,68 @@
...
-;environment=KEY=value       ; (key value pairs to add to environment)
 
 [program:mosquitto]
 command=/usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf
+user=mosquitto
 priority=100
-stdout_logfile=/var/hub-container/log/mosquitto-stdout.log
-stderr_logfile=/var/hub-container/log/mosquitto-stderr.log
+stdout_logfile=/var/hub-container/log/mosquitto.log
+stderr_logfile=/var/hub-container/log/mosquitto.log
```
Comment (muhlpachr) [inline etc/supervisord.conf]: Why not redirect_stderr ?

File: etc/supervisord.conf [hunk 2]
```diff
@@ -1,85 +1,68 @@
...
 
 [program:bridge]
-autostart=true
```
Comment (muhlpachr) [inline etc/supervisord.conf]: I prefere to leave default parameter mentioned to allow easy change of configuration (to allow user identify directive without study of supervisord documentation.

File: etc/supervisord.conf [hunk 3]
```diff
@@ -1,85 +1,68 @@
...
 nodaemon=true
-logfile=/var/hub-container/log/supervisord.log ; (main log file;default $CWD/supervisord.log)
-logfile_maxbytes=1MB       ; (max main logfile bytes b4 rotation;default 50MB)
-logfile_backups=1          ; (num of main logfile rotation backups;default 10)
-loglevel=info                ; (log level;default info; others: debug,warn,trace)
+childlogdir=/var/hub-container/log
+logfile=/var/hub-container/log/supervisord.log
+logfile_maxbytes=1MB  ; max main logfile bytes before rotation
+logfile_backups=1  ; num of main logfile rotation backups
+loglevel=info
```
Comment (muhlpachr) [inline etc/supervisord.conf]: Why to remove helper comment for other levels ?
That could easy edit configuration with levels helpers for troubleshooting.

File: etc/supervisord.conf [hunk 4]
```diff
@@ -1,85 +1,68 @@
...
 pidfile=/run/supervisord.pid
-childlogdir=/var/hub-container/log   ; ('AUTO' child log dir, default $TEMP)
-;environment=KEY=value       ; (key value pairs to add to environment)
```
Comment (muhlpachr) [inline etc/supervisord.conf]: It is intetional to leave some helpers for expected configuration directives as comments.

File: etc/supervisord.conf [hunk 5]
```diff
@@ -1,85 +1,68 @@
...
 priority=200
-stdout_logfile=/var/hub-container/log/bridge-stdout.log
-stderr_logfile=/var/hub-container/log/bridge-stderr.log
+stdout_logfile=/var/hub-container/log/bridge.log
+stderr_logfile=/var/hub-container/log/bridge.log
 
 [program:bridge-simple]
 autostart=false
 command=/usr/bin/bridge-simple mqtt://localhost nodes/bridge/0
+user=nobody
```
Comment (muhlpachr) [inline etc/supervisord.conf]: Is it final or intermediate precaution to run various components under same user privileges ?

File: etc/supervisord.conf [hunk 6]
```diff
@@ -1,85 +1,68 @@
...
 
 [program:blynk-client]
-autostart=true
 command=/var/hub-container/plugins/blynk-vhw/blynk-client.py
+user=nobody
 priority=500
-stdout_logfile=/var/hub-container/log/blynk-client-stdout.log
-stderr_logfile=/var/hub-container/log/blynk-client-stderr.log
-
-;environment=A=1,B=2           ; process environment additions (def no adds)
```
Comment (muhlpachr) [inline etc/supervisord.conf]: I prefer to leave those comments as helpers.

File: etc/supervisord.conf [hunk 7]
```diff
@@ -1,85 +1,68 @@
 [supervisord]
 nodaemon=true
-logfile=/var/hub-container/log/supervisord.log ; (main log file;default $CWD/supervisord.log)
-logfile_maxbytes=1MB       ; (max main logfile bytes b4 rotation;default 50MB)
-logfile_backups=1          ; (num of main logfile rotation backups;default 10)
-loglevel=info                ; (log level;default info; others: debug,warn,trace)
+childlogdir=/var/hub-container/log
+logfile=/var/hub-container/log/supervisord.log
+logfile_maxbytes=1MB  ; max main logfile bytes before rotation
```
Comment (muhlpachr) [inline etc/supervisord.conf]: Just curious why to change original config template comments and remove brackets ?

File: etc/supervisord.conf [hunk 8]
```diff
@@ -1,85 +1,68 @@
...
-
-;[group:thegroupname]
-;programs=progname1,progname2  ; each refers to 'x' in [program:x] definitions
-;priority=999                  ; the relative start priority (default 999)
+stdout_logfile=/var/hub-container/log/blynk-client.log
+stderr_logfile=/var/hub-container/log/blynk-client.log
 
 [supervisorctl]
-serverurl=unix:///run/supervisord.sock ; use a unix:// URL  for a unix socket
+serverurl=unix:///run/supervisord.sock  ; use a unix:// URL  for a unix socket
```
Comment (muhlpachr) [inline etc/supervisord.conf]: UNIX socket has issues in RPi Docker, we should switch to TCP socket (could be done in another pull).

File: etc/supervisord.conf (hunk 1 above)
Comment (jirutka) [inline etc/supervisord.conf]: Good idea! I I misunderstood documentation for `redirect_stderr`.

File: etc/supervisord.conf (hunk 2 above)
Comment (jirutka) [inline etc/supervisord.conf]: Okay.

File: etc/supervisord.conf (hunk 5 above)
Comment (jirutka) [inline etc/supervisord.conf]: It’s final. All the programs that run as user nobody don’t write to file system or read any sensitive files, they are not directly exposed to outside world (to be correct, this applies to clown-talk-ws-server only partially), and they are made by us. I think that it’s unnecessary to create a separate user or even users for them.

File: etc/supervisord.conf (hunk 6 above)
Comment (jirutka) [inline etc/supervisord.conf]: I personally prefer to not pollute configuration files with blocks of irrelevant comments. Moreover, these comments are quite incomprehensible. For example, what should mean “number of bytes in 'capturemode'”?!

File: etc/supervisord.conf (hunk 8 above)
Comment (jirutka) [inline etc/supervisord.conf]: What issues exactly? It works for me.

File: etc/supervisord.conf (hunk 4 above)
Comment (jirutka) [inline etc/supervisord.conf]: We don’t utilize `childlogdir`, so it’s useless here. I actually wanted to use it, but it seems that there’s no way how to customize default file names that it generates; the default ones are quite inconvenient.

**Comment by jirutka** [comment]
Just a reminder, **do not** merge it until I rebase it **after** the review is finished.

**Comment by jirutka** [comment]
Rebased, updated changelog, ready to merge.

**Pull request merged**